openpype/hosts/nuke/api/workfile_template.py

```diff
@@ -0,0 +1,337 @@
+        '''
+
+        t_nodes = self._get_op_template_nodes()
+        t_creators = [node for node in t_nodes
+                      if node['is_op_creator'].value()]
+        asset = str(os.environ["AVALON_ASSET"])
+        for temp_creator in t_creators:
+            load_create = str(temp_creator['loaderCreator'].value())
+            subset = str(temp_creator['subset'].value())
+
+            # Get appropriate plugin class
+            creator_plugin = None
+            for creator in discover_legacy_creator_plugins():
+                if str(creator.__name__) != load_create:
+                    continue
+                creator_plugin = creator
+
+            if creator_plugin:
+                creator_plugin(subset, asset).process()
+            else:
+                log.warning('Creator plugin {} not found'.format(load_create))
+
+    def _template_connect(self):
+        '''
+        Swaps OP loaders or creators with corresponding template nodes,
+        delete Template nodes
+        '''
+
+        op_nodes = self._get_op_nodes(selected_only=False)
+        template_nodes = self._get_op_template_nodes()
+
+        for one_template in template_nodes:
+            load_create = str(one_template['loaderCreator'].value())
+            subset = str(one_template['subset'].value())
+            is_op_creator = bool(one_template['is_op_creator'].value())
+            knob_dict = {}
+            try:
+                knob_dict = ast.literal_eval(
+                    str(one_template['other'].value()))
+            except ValueError:
+                log.warning('Temp node {} knobs failed to parse.'.format(
+                    one_template['name'].value()))
+
+            target_node = None
+            if is_op_creator:
+                # it is a creator, compare subsets
+                for node in op_nodes:
+                    try:
+                        if subset == str(node['avalon:subset'].value()):
+                            target_node = node
+                            break
+                    except NameError:
+                        pass
+            else:
+                # it is a loader, compare subset and loader name
+                for node in op_nodes:
+                    try:
+                        if subset == str(node['avalon:name'].value()):
+                            node_loader = str(node['avalon:loader'].value())
+                            if load_create == node_loader:
+                                target_node = node
+                                break
+                    except NameError:
+                        pass
+
+            if target_node:
+                self._swap_nodes(one_template, target_node)
+                for knob_name, knob in knob_dict.items():
+                    try:
+                        knob_value = str(knob['value'])
+                        knob_type = int(knob['type'])
+                        if knob_type == 3:
```

Comment:
If knob types were aligned with Setting's style you could rather use `openpype\hosts\nuke\api\lib.py:set_node_knobs_from_settings` for setting knob values.